src/host_toolchains.py

```diff
@@ -0,0 +1,87 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#   Copyright 2015 WebAssembly Community Group participants
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import json
+import os
+
+import proc
+
+WORK_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'work')
+CR_BUILD_DIR = os.path.join(WORK_DIR, 'build')
+SETUP_TOOLCHAIN = os.path.join(CR_BUILD_DIR, 'toolchain', 'win', 'setup_toolchain.py')
+V8_SRC_DIR = os.path.join(WORK_DIR, 'v8', 'v8')
+VS_TOOLCHAIN = os.path.join(V8_SRC_DIR, 'gypfiles', 'vs_toolchain.py')
+WIN_TOOLCHAIN_JSON = os.path.join(V8_SRC_DIR, 'gypfiles', 'win_toolchain.json')
+
+os.environ['GYP_MSVS_VERSION'] = '2015'
+
+def SyncPrebuiltClang(name, src_dir, git_repo):
+  """Update the prebuilt clang toolchain used by chromium bots"""
+  tools_clang = os.path.join(src_dir, 'tools', 'clang')
+  if os.path.isdir(tools_clang):
+    print 'Prebuilt Chromium Clang directory already exists'
+  else:
+    print 'Cloning Prebuilt Chromium Clang directory'
+    Mkdir(src_dir)
+    Mkdir(os.path.join(src_dir, 'tools'))
+    Git(['clone', git_repo, tools_clang])
+  Git(['fetch'], cwd=tools_clang)
+  proc.check_call(
+      [os.path.join(tools_clang, 'scripts', 'update.py')])
+  assert os.path.isfile(CC), 'Expect clang at %s' % CC
+  assert os.path.isfile(CXX), 'Expect clang++ at %s' % CXX
+  return ('chromium-clang', tools_clang)
+
+
+def SyncWinToolchain():
+  """Update the VS toolchain used by Chromium bots"""
+  proc.check_call([VS_TOOLCHAIN, 'update'])
+
+
+def GetEnv(dir):
+  """ Return the configured environment block as a python dict."""
+  env = {}
+  with open(os.path.join(dir, 'environment.x64'), 'rb') as f:
+    entries = f.read().split('\0')
+    for e in entries:
+      if not e:
+        continue
+      var, val = e.split('=', 1)
+      env[var] = val
+  return env
+
+
+def SetUpEnv(outdir):
```

Comment:
Seems like this is Windows-specific? Probably should be clearer in the function name.